### ISSUE TITLE
Buffers and working with rate chapter should mention fusion

### ISSUE THREAD

**Issue by Mihai238** [akka/akka#20017]
Hi,

According to http://doc.akka.io/docs/akka/2.4.2/java/stream/stream-rate.html akka streams runs stages asynchronously, starting to handle one element immediately after the previous one was delivered to its downstream.

Following code:

```java
public static void main(String[] args) throws InterruptedException {
        ActorSystem system = ActorSystem.create();
        ActorMaterializer mat = ActorMaterializer.create(system);

        Source.from(Arrays.asList(1, 2, 3))
                .map(i -> {System.out.println("A: " + i); return i;})
                .map(i -> {System.out.println("B: " + i); return i;})
                .map(i -> {System.out.println("C: " + i); return i;})
                .runWith(Sink.<Integer>ignore(), mat);
    }
```

produces always the following output on my computer:

```
A: 1
B: 1
C: 1
A: 2
B: 2
C: 2
A: 3
B: 3
C: 3
```

Same output is produced also if I add a `Thread.sleep(1000)` statement in the last `map` operation. There is no `application.conf` file in my classpath. Am I missing something?

Additional information:
Akka Stream version: akka-stream_2.11:2.4.2
Java Version: openjdk 1.8.0_74

**Comment by drewhk** [comment]
By default stages are fused, you need to mark your stages async explicitly, see http://doc.akka.io/docs/akka/2.4.2/scala/stream/stream-flows-and-basics.html#Operator_Fusion for details.

**Comment by Mihai238** [comment]
Oh, I missed that. Thank you! Nevertheless, it would be great if this information would also be stated in `Buffers and working with rate` chapter.

**Comment by drewhk** [comment]
Ok, I renamed the ticket and reopened.

**Comment by johanandren** [comment]
Was actually fixed in 2.4.3 but forgot to mark ticket as such.
